File: docs/static/examples/src/flappy-bird/c++/flappy_bird.cpp
#include "p8g.hpp"

#include <math.h>
#include <vector>

using namespace p8g;

Image background_day = loadImage("background-day.png");
Image base = loadImage("base.png");
std::vector<Image> yellowbird = { loadImage("yellowbird-downflap.png"), loadImage("yellowbird-midflap.png"), loadImage("yellowbird-upflap.png") };

void p8g::draw() {
    image(background_day, 0, 0);
    image(base, -(millis() / 10) % width, height - base.height);
    image(base, -(millis() / 10) % width + base.width, height - base.height);
    imageMode(CENTER);
    image(yellowbird[(millis() / 100) % yellowbird.size()], width / 2, height / 2 + 2 * sinf(millis() / 100.f));
}

int main() {
    run(288, 512, "Flappy Bird");
}

File: docs/static/examples/src/flappy-bird/java/FlappyBird.java
import io.github.bernhardfritz.p8g.Image;
import io.github.bernhardfritz.p8g.Sketch;

import java.util.List;

public class FlappyBird extends Sketch {
    Image backgroundDay = loadImage("background-day.png");
    Image base = loadImage("base.png");
    List<Image> yellowbird = List.of(loadImage("yellowbird-downflap.png"), loadImage("yellowbird-midflap.png"), loadImage("yellowbird-upflap.png"));

    @Override
    protected void draw() {
        image(backgroundDay, 0, 0);
        image(base, -(millis() / 10) % width, height - base.height);
        image(base, -(millis() / 10) % width + base.width, height - base.height);
        imageMode(CENTER);
        image(yellowbird.get((millis() / 100) % yellowbird.size()), width / 2, height / 2 + 2 * (float) Math.sin(millis() / 100.f));
    }

    public static void main(String[] args) {
        Sketch sketch = new FlappyBird();
        sketch.run(288, 512, "Flappy Bird");
    }
}
